ydb/src/client_topic/list_types.rs

```diff
@@ -0,0 +1,107 @@
+use crate::errors;
+use crate::grpc_wrapper::raw_topic_service::common::codecs::{RawCodec, RawSupportedCodecs};
+use crate::grpc_wrapper::raw_topic_service::common::consumer::RawConsumer;
+use crate::grpc_wrapper::raw_topic_service::common::metering_mode::RawMeteringMode;
+use std::collections::HashMap;
+use std::time::Duration;
+
+#[derive(Clone, Default, PartialEq, Eq)]
+pub struct Codec {
+    pub code: i32,
+}
+
+impl Codec {
+    pub const RAW: Codec = Codec { code: 1 };
+    pub const GZIP: Codec = Codec { code: 2 };
+    pub const LZOP: Codec = Codec { code: 3 };
+    pub const ZSTD: Codec = Codec { code: 4 };
+
+    pub fn is_custom(&self) -> bool {
+        self.code >= 10000 && self.code < 20000
+    }
+}
+
+#[derive(Clone, Default)]
+pub struct SupportedCodecs {
+    pub codecs: Vec<Codec>,
+}
+
+impl TryFrom<Codec> for RawCodec {
```

Comment:
It no need error for conversion - user can set any codec. For example if YDB will publish "reverse" codec as well-known const = 5 and user have implementation for this codec (same way as for any custom codec). No reason for deny to use it.

It can implement simple From and direct copy codec value to Raw format.

Raw can detect well-known constants if someself (if need).